lib/api.js

```diff
@@ -756,8 +766,22 @@ API.prototype.savePreferences = function(preferences, cb) {
 
 
 API.prototype._computeProposalSignature = function(args) {
-  $.shouldBeNumber(args.amount);
-  var hash = WalletUtils.getProposalHash(args.toAddress, args.amount, args.message, args.payProUrl);
+  var hash;
+  if (args.outputs) {
+    $.shouldBeArray(args.outputs);
+    // should match bws server createTx
+    var proposalHeader = {
+      outputs: _.map(args.outputs, function(output) {
+        return _.pick(output, ['toAddress', 'amount', 'message']);
+      }),
+      message: args.message,
+      payProUrl: args.payProUrl
+    };
+    hash = WalletUtils.getProposalHash(proposalHeader);
+  } else {
+    $.shouldBeNumber(args.amount);
```

Comment:
If this test is needed (I'm not so sure), it should also be performed on the `amount` attribute of the `outputs` array?